Aquality.Selenium.Core/src/Aquality.Selenium.Core/Elements/ElementFactory.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Aquality.Selenium.Core.Elements.Interfaces;
+using Aquality.Selenium.Core.Localization;
+using Aquality.Selenium.Core.Waitings;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.PageObjects;
+
+namespace Aquality.Selenium.Core.Elements
+{
+    /// <summary>
+    /// Factory that creates elements.
+    /// </summary>
+    public class ElementFactory : IElementFactory
+    {
+        private const string ByXpathIdentifier = "By.XPath";
+
+        public ElementFactory(ConditionalWait conditionalWait, IElementFinder elementFinder, LocalizationManager localizationManager)
+        {
+            ConditionalWait = conditionalWait;
```

Comment:
why do not we get all these objects from container?
at least 'ConditionalWait' here looks as strange as LocalizationManager

though maybe it is about core-flexibility and customizability

Reply:
> why do not we get all these objects from container

 - We DO resolve them via container, and the constructor injection is the right way to resolve the dependencies in dependencies when we use the DI container.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection
Also we SHOULD resolve all dependencies that we use the same way. There no need to hide this, because it's our particular implementation. We should keep it as transparent as possible.